packages/grid/_modules_/grid/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -54,6 +37,70 @@ export const cleanFilterItem = (item: GridFilterItem, apiRef: GridApiRef) => {
   return cleanItem;
 };
 
+const filterModelDisableMultiColumnsFilteringWarning = buildWarning([
+  'MUI: The `filterModel` can only contain a single item when `prop.disableMultipleColumnsFiltering` is set to `true`.',
+  'If you are using the `DataGrid` community version, this property is always `true`.',
+]);
+
+const filterModelMissingItemIdWarning = buildWarning(
+  "MUI: The 'id' field is required on `filterModel.items` when you use multiple filters.",
+);
+
+export const sanitizeFilterModel = (
+  model: GridFilterModel,
+  disableMultipleColumnsFiltering: boolean,
+  apiRef: GridApiRef,
+) => {
+  if (model.items.length > 1) {
+    if (disableMultipleColumnsFiltering) {
+      filterModelDisableMultiColumnsFilteringWarning();
+
+      return {
+        ...model,
+        items: [model.items[0]],
+      };
+    }
+
+    let hasItemsWithoutIds = false;
+    let hasItemWithoutOperator = false;
+
+    for (let i = 0; i < model.items.length; i += 1) {
+      const item = model.items[i];
+
+      if (item.id == null) {
+        hasItemsWithoutIds = true;
+      }
+
+      if (item.operatorValue == null) {
+        hasItemWithoutOperator = true;
+      }
+    }
+
+    if (hasItemsWithoutIds) {
+      filterModelMissingItemIdWarning();
+    }
+
+    if (hasItemWithoutOperator || hasItemsWithoutIds) {
```

Comment:
Should we make `operatorValue` required in v6? If yes, then makes sense to also add a warning for it.

Reply:
I would say yes, I don't like when we are doing magic like adding a default operator.

@alexfauquette if you have a thought on this

Reply:
Yes, we could make it required.

If it does not cause a problem, I would be in favor of keeping the default operator, just to save the ass of developers that miss use the grid in production

Reply:
As I see it, it would be like the ID

- The typing says it's required
- If none is provided, we fire a warning and provide a default behavior

Reply:
I added a line in Preparing v6 such that we don't forget it

Reply:
I added a warning + doc line + TODO